packages/ui/src/providers/delete-modal.provider.test.tsx

```diff
@@ -0,0 +1,52 @@
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import { useContext } from 'react';
+import { DeleteModalContextProvider, DeleteModalContext } from './delete-modal.provider';
+
+let deleteConfirmationResult: boolean | undefined;
```

Comment:
What about setting this variable as `undefined` in a `beforeEach` inside of the `describe` section? This way we ensure that no matter the test order, the result can be achieved properly